src/varname.jl

```diff
@@ -2,42 +2,41 @@ using Setfield
 using Setfield: PropertyLens, ComposedLens, IdentityLens, IndexLens, DynamicIndexLens
 
 """
-    VarName{sym}(indexing::Tuple=())
+    VarName{sym}(lens::Lens=IdentityLens())
 
-A variable identifier for a symbol `sym` and indices `indexing` in the format
-returned by [`@vinds`](@ref).
+A variable identifier for a symbol `sym` and lens `lens`.
 
 The Julia variable in the model corresponding to `sym` can refer to a single value or to a
-hierarchical array structure of univariate, multivariate or matrix variables. The field `indexing`
+hierarchical array structure of univariate, multivariate or matrix variables. The field `lens`
 stores the indices requires to access the random variable from the Julia variable indicated by `sym`
-as a tuple of tuples. Each element of the tuple thereby contains the indices of one indexing
+as a tuple of tuples. Each element of the tuple thereby contains the indices of one lens
 operation.
 
-`VarName`s can be manually constructed using the `VarName{sym}(indexing)` constructor, or from an
-indexing expression through the [`@varname`](@ref) convenience macro.
+`VarName`s can be manually constructed using the `VarName{sym}(lens)` constructor, or from an
+lens expression through the [`@varname`](@ref) convenience macro.
 
 # Examples
 
 ```jldoctest; setup=:(using Setfield)
 julia> vn = VarName{:x}(Setfield.IndexLens((Colon(), 1)) ∘ Setfield.IndexLens((2, )))
 x[:,1][2]
 
-julia> vn.indexing
+julia> getlens(vn)
 (@lens _[Colon(), 1][2])
 
 julia> @varname x[:, 1][1+1]
 x[:,1][2]
 ```
 """
 struct VarName{sym,T<:Lens}
-    indexing::T
+    lens::T
 
-    function VarName{sym}(indexing=IdentityLens()) where {sym}
+    function VarName{sym}(lens=IdentityLens()) where {sym}
         # TODO: Should we completely disallow or just `@warn` of limited support?
```

Comment:
IMO we disallow for now, and then we can consider enabling dynamic lenses if we have a good use-case for it.

Reply:
Sounds good.